### PR TITLE
New EMS calling points BeginZoneTimestepBeforeInitHeatBalance and After

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Add new EMS calling point BeginZoneTimestepBeforeInitHeatBalance so that actuators related to constructions, shading, and windows can be activated prior to the shading, daylighting, and surface heat balance calculations. Follow up to #6801.
Also add new calling point BeginZoneTimestepAfterInitHeatBalance.

Partially addresses #6626.

Moves toward completing #4266, #3654, and maybe others.

Also happens to address #7179, EMS for internal gains.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex
Comment (Myoldmopar) [inline doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex]: This will be a nice addition.

File: doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex
Comment (Myoldmopar) [inline doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex:11]: It's great to clarify the use case for this calling point separate from the new one.

Comment (Myoldmopar) [reply doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex:11]: Does the IO ref need any accompanying changes?

File: src/EnergyPlus/DataGlobals.cc
Comment (Myoldmopar) [inline src/EnergyPlus/DataGlobals.cc:161]: Looks good.

File: testfiles/EMSThermochromicWindow.idf
Comment (Myoldmopar) [inline testfiles/EMSThermochromicWindow.idf]: Will you expect any diffs because of this?

**Review by Myoldmopar — APPROVED** [review]

File: doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex
Comment (mjwitte) [reply doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex:11]: Oh - yes, it does. Will add.

File: testfiles/EMSThermochromicWindow.idf
Comment (mjwitte) [reply testfiles/EMSThermochromicWindow.idf]: Yes, but now I wonder if I should leave that for a different PR and just add the calling point here.

Comment (Myoldmopar) [reply testfiles/EMSThermochromicWindow.idf]: Nah, CI has already started working on it.  Let it finish as-is so it can merge in.  I'll anticipate diffs in these two files and if we need to investigate further, we'll do it after we freeze the IDD.

File: doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex
Comment (Myoldmopar) [reply doc/ems-application-guide/src/ems-calling-points/begin-timestep-before-predictor.tex:11]: Oh, hmm.  OK, well if you make this change on the docs then I guess I'll defer to you whether you want to revert the changes to the test files below.  I don't care either way.  At some point we'll want to change the test file(s) to have this new calling point, but because of my concern with getting the IDD frozen, I'm indifferent on whether it is with this PR or not.

File: doc/input-output-reference/src/overview/group-energy-management-system-ems.tex
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-energy-management-system-ems.tex:119]: :+1:

File: doc/input-output-reference/src/overview/group-energy-management-system-ems.tex
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-energy-management-system-ems.tex:122]: :+1:

**Comment by Myoldmopar** [comment]
I do **not** know what CI is barking about on the unit test platform.  I just built it and ran it fine, though it was in release mode.  Lemme try it in debug mode and make sure it's OK.

But CI didn't complain about a failing build, or anything.  So I'm not sure.

**Comment by Myoldmopar** [comment]
> Lemme try it in debug mode and make sure it's OK.

Built it in debug mode and re-ran the unit tests.  Everything is fine.  I'm going to chock that up to a fluke for right now.

**Comment by mjwitte** [comment]
@Myoldmopar I'm thinking I should revert the two test files just to make this cleaner. Yea or nay?
@XuanLuoLBL will look at the hybrid test file failures (duplicate headers).

**Comment by Myoldmopar** [comment]
I think yes, go ahead and revert those out.  It will give CI a chance to re-run the unit test check on here which I would love to see be green again.

**Comment by Myoldmopar** [comment]
This looks clean, just some reordered MTD and RDD files, but no actual diffs.  This should be able to drop in shortly.

**Comment by Myoldmopar** [comment]
@mjwitte The VM was running the very last integration test file (VSWater...) and it was going to take a full hour to try to run that file before it ultimately times out anyway.  I went ahead and killed it.  This will show up as a test failure when CI reports here soon, but it saved a chunk of testing time.  I'll merge this in once CI posts here.

**Comment by Myoldmopar** [comment]
@mjwitte I'm not going to try to close those issues that are mentioned at the top of this PR today.  From a quick glance, I'm guessing it closes either 1 of the 3 or 2 of the 3.  If you know for sure that it does *not* close any of them, feel free to edit the comment and I'll be happy to review the expected fixed ones soon.

**Comment by mjwitte** [comment]
Well, it doesn't close anything unless the new calling points are actually used, so don't close any of them.

File: doc/input-output-reference/src/overview/group-energy-management-system-ems.tex
Comment (DuzzleJay) [reply doc/input-output-reference/src/overview/group-energy-management-system-ems.tex:119]: Hi Myoldmopar, hope you're doing well. I’m reaching out with a question about an EnergyPlus EMS issue I’ve been stuck on, and I’d really appreciate your insight.
I’m running a continuous simulation in EnergyPlus 9.4, using EMS + Schedule:File to dynamically switch between EC window constructions (each with unique solar and visible transmittance) at runtime. The goal is to capture both thermal and lighting performance for total energy optimization, so I need both HVAC loads and lighting consumption to respond to the changing window states.
I’ve confirmed the setup works for thermal performance: cooling/heating loads change correctly with the window states, proving the EMS construction switching is working. Static runs with each fixed construction also show expected differences in daylight illuminance and lighting energy, so the daylighting/lighting controls themselves are set up properly.
The issue is that in the continuous EMS runs, daylight illuminance and lighting electricity consumption do NOT update with the window state—they stay fixed at the value calculated for the initial construction, even when the window switches to a state with very different visible transmittance. I’ve already tested all three relevant EMS calling points (BeginTimestepBeforePredictor, BeginZoneTimestepBeforeInitHeatBalance, BeginZoneTimestepAfterInitHeatBalance), but none triggered the daylighting module to recalculate, even though thermal loads still responded correctly.
Do you know of any workaround or specific setting to make EnergyPlus recalculate illuminance and lighting energy when window constructions are changed via EMS at runtime, while keeping the simulation continuous to preserve thermal inertia?
Any guidance would be really helpful. Thank you so much!

**Pull request merged**
